File: Algorithm/Programers/lv01/p12977/programers_12977.js
// https://programmers.co.kr/learn/courses/30/lessons/12977

function solution(nums) {
    let nums_len = nums.length;
    let answer = 0;

    for (let i = 0; i < nums_len - 2; i++) {
        for (let j = i + 1; j < nums_len - 1; j++) {
            for (let k = j + 1; k < nums_len; k++) {
                if (lsPrime(nums[i], nums[j], nums[k])) {
                    answer++;
                }
            }
        }
    }
    return answer;
}

function lsPrime(a, b, c) {
    let sum = a + b + c;
    for (let i = sum - 1; i > 1; i--) {
        if (sum % i == 0) {
            return false;
        }
    }
    return true;
}

File: Algorithm/Programers/lv01/p12977/programers_12977.c
// https://programmers.co.kr/learn/courses/30/lessons/12977

#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int nums[], size_t nums_len) {
    int answer = 0;
    for(int i=0; i<nums_len-2; i++){
        for(int j=i+1; j<nums_len-1; j++){
            for(int k=j+1; k<nums_len; k++){
                if(lsPrime(nums[i], nums[j], nums[k])){
                    answer++;
                }
            }
        }
    }
    return answer;
}

int lsPrime(int a, int b, int c){
    int sum = a + b + c;
    for(int i=sum-1; i>1; i--){
        if(sum%i == 0){
            return false;
        }
    }
    return true;
}
